pkg/kubectl/cmd/edit.go

```diff
@@ -273,7 +281,11 @@ func RunEdit(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args
 				meta.SetList(updates.Object, mutatedObjects)
 			}
 
-			err = visitToPatch(originalObj, updates, mapper, resourceMapper, encoder, out, errOut, defaultVersion, &results, file)
+			if isUsedByEditCmd {
+				err = visitToPatch(originalObj, updates, mapper, resourceMapper, encoder, out, errOut, defaultVersion, &results, file)
+			} else {
+				err = visitToCreate(updates, mapper, resourceMapper, out, errOut, defaultVersion, &results, file)
```

Comment:
What if we have more commands using `edit` in the future? We may need to update all `if`s and it's easy to miss one.